### DISCUSSION TITLE
API question: How to get the ordinal transactions of a certain block?

### DISCUSSION THREAD

**Discussion by nosuic** [ordinals/ord#3830] [Q&A]
Hi all,

Thanks to the invaluable [help](https://github.com/ordinals/ord/discussions/3663) of @cryptoni9n I have been able to extract pretty much all data (including the recursive modules) of specific inscriptions.

I wanted to ask something that may be trivial; Although I am able to retrieve all the inscriptions of a certain block, I cannot figure out a way to retrieve the block's ordinal transactions.

For example: The following call on ordinals.com returns 4,250 transactions, however when making the exact same call using the ord API, no transactions are returned:

https://ordinals.com/block/00000000000000000002c924a78c32e02a16183aa7179304ce5ff3429c8747fc

Can anyone please recommend a way:
1.  To get all inscription-related transactions for a certain block?
2. Or otherwise, to get the transactions (sales and transfers) of a certain inscription?

Thanks so much!

**Comment by cryptoni9n** [comment]
Hi @nosuic,

To get all inscription ids for the block: `curl -s -X GET http://localhost/inscriptions/block/844644 | grep -oP '(?<=src=/preview/)[^"]+' | awk -F'[<>]' '{print $1}'`

To get the tx of a certain inscription (ex: 71012647 ): `curl -s -X GET http://localhost/inscription/71012647 | grep -oP '(?<=<dd class=monospace>)[^<]+' | head -n 1`

I hope this helps.

**Reply by nosuic** [reply]
Hi @cryptoni9n,

Thanks so much for your response! 

When running the second command your shared (i.e. for getting the transactions of a certain inscription) it only seems to return the inscription ID. The command used:
```
eg@eg-desktop:~$  curl -s -X GET http://localhost/inscription/71012647 | grep -oP '(?<=<dd class=monospace>)[^<]+' | head -n 1
f66b6f9954121dfb5d52398aa92b48153463c931d4f533c9ba06a2736dd164e8i0
```
Any ideas why is that and what adjustments we should make to get a list of the inscription's transactions instead?

Thanks again!

**Reply by cryptoni9n** [reply]
oh, sorry about that.  the inscription id is very similar to the reveal tx id.  here's the command to return the reveal tx:

```rust
curl -s -X GET http://localhost/inscription/71012647 | grep -oP '(?<=<a class=monospace href=/tx/)[^<]+' | sed 's/.*>//'
```
